File: experiments/source-mapping/runner.js
import { chromium } from 'playwright';
import fs from 'fs/promises';

async function scan() {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  console.log('Navigating to http://localhost:3001');
  await page.goto('http://localhost:3001', { waitUntil: 'networkidle' });

  await page.addScriptTag({ path: './node_modules/axe-core/axe.min.js' });

  console.log('Running Axe...');
  const results = await page.evaluate(async () => {
  return await axe.run();
  });

  const findings = [];

  for (const violation of results.violations) {
  for (const node of violation.nodes) {
    findings.push({
    ruleId: violation.id,
    impact: violation.impact,
    htmlSnippets: [node.html],
    selectors: node.target,
    failureSummary: node.failureSummary
    });
  }
  }

  await fs.writeFile('findings.json', JSON.stringify(findings, null, 2));
  console.log(`Saved ${findings.length} findings to findings.json`);

  await browser.close();
}

scan().catch(console.error);

File: experiments/source-mapping/runner.py
import asyncio
from playwright.async_api import async_playwright
import json
import os

async def scan():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        await page.goto("http://localhost:3001")
        
        # Inject axe-core
        await page.add_script_tag(path="node_modules/axe-core/axe.min.js")
        
        # Run axe
        results = await page.evaluate("async () => await axe.run()")
        
        findings = []
        for violation in results.get("violations", []):
            for node in violation.get("nodes", []):
                findings.append({
                    "ruleId": violation["id"],
                    "impact": violation.get("impact"),
                    "html": node.get("html"),
                    "target": node.get("target"),
                    "failureSummary": node.get("failureSummary")
                })
        
        with open("findings.json", "w") as f:
            json.dump(findings, f, indent=2)
            
        await browser.close()
        print(f"Saved {len(findings)} findings.")

if __name__ == "__main__":
    asyncio.run(scan())
